examples/utils.py

```diff
@@ -118,18 +137,28 @@ def __init__(self, save_path, metric, patience):
     self.patience_count = 0
 
   def __call__(self, model, current_score):
+    from deepchem.models import TorchModel
     if self.mode == 'higher' and current_score > self.best_score:
       self.best_score = current_score
       self.patience_count = 0
-      torch.save(model.model.state_dict(), self.save_path + '/early_stop.pt')
+      if type(model).__bases__[0] == TorchModel:
+        torch.save(model.model.state_dict(), self.save_path + '/early_stop.pt')
+      else:  # KerasModel
+        model.model.save(self.save_path + '/early_stop')
     elif self.mode == 'lower' and current_score < self.best_score:
       self.best_score = current_score
       self.patience_count = 0
-      torch.save(model.model.state_dict(), self.save_path + '/early_stop.pt')
+      if type(model).__bases__[0] == TorchModel:
+        torch.save(model.model.state_dict(), self.save_path + '/early_stop.pt')
+      else:  # KerasModel
+        model.model.save(self.save_path + '/early_stop')
     else:
       self.patience_count += 1
 
     return self.patience_count == self.max_patience
 
   def load_state_dict(self, model):
```

Comment:
I wonder if it's better to use a single API for `load_state_dict` and `load_keras_model`.

Reply:
I know that all classes that inherit from `KerasModel` have a `restore()` method and do automatic checkpointing, so I'm actually not even using the `model.model.save()` for anything here. I'm not sure if `TorchModel` has a standard way of restoring, but that could be a good thing to standardize in a future PR.

Reply:
Got it.